pages/blog/_posts/2020-02-20-Extending-Kubernetes-Static-IP-Routing.md

```diff
@@ -0,0 +1,82 @@
+downsides.
+
+### Adding a scheduler extender
+The kubernetes scheduler checks for certain requirements before it schedules a pod onto a node. Some of these requirements
+are hard requirements, like cpu, memory and number of pods. Other requirements are more soft, like if the pods are allowed to
+be packed together or in which AZ they are gonna run. All of those requirements are collected and points given to each
+node on how good they meet the requirements. The Node that fits best, gets chosen.
+
+All of that are things the scheduler will do for you automatically, however it is also possible to give the cluster a 
+`KubeSchedulerConfiguration` object that will tell the scheduler to also reach out to a service for additional point scoring.
+The SchedulerConfiguration is a JSON file and for further explanations, please have a look at this 
+excellent [blog post](https://developer.ibm.com/articles/creating-a-custom-kube-scheduler/).
+
+In our case, we could have written a service that checks which IP Addresses are assigned to the Nodes and moved the Pods
+onto those nodes. This would have required us to make sure that those Nodes had all needed IP-Addresses all the time.
+That sounded not very enticing when doing cluster upgrades as it would have needed to be at least a semi-manual process.
+
+### Creating an operator
+An Operator/Controller on the other hand is a component observing resources and then try to create the described resources.
+The difference between a controller and an operator is basically that an operator is handling the lifecycle of an
+application, whereas a controller may control a specific resource that is not associated with a specific application.
+They both use the controller pattern though and both can be implemented with the same toolset, so for simplicity sake in the
+context of this article, we will consider them to be equal.
+
+Operators usually consists at least out of a CRD(CustomResourceDefinition), an Event Listener and a Reconciliation Loop.
+The CRD defines the object that shall be created, e.g. a Redis deployment. The whole description on what this deployment
+needs to look like and all its abilities need to be defined in the CRD. The Operator will create an Event Listener for that
+CRD as the primary resource and additional event listeners for the secondary resource (most likely pods in this example).
+The Event Listener will let the Reconciliation Loop know once a CRUD operation has been requested on either the primary or
+secondary resource. The Loop will then try to bring the CRD into the desired state, depending on what operation has been
+requested. So in the Redis example it will either create the pods, update the pods, in the case of an upgrade it might blue/green
+the pods or delete the pods. Basically things a human operator would do in this case, just in programmatic form, taking it
+from a declarative form into existing resources.
```

Comment:
There's a lot of information in this paragraph - maybe split it up a bit?

Reply:
I tried splitting it up, but did not succeed ... did sprinkle a little bit of highlighting on several words to make it easier to follow though